code/datums/proximity_monitor/fields/bubble.dm

```diff
@@ -35,6 +38,27 @@
 	my_movable = null
 	return ..()
 
+/datum/proximity_monitor/advanced/bubble/proc/set_glide_host(atom/movable/new_glide_host)
+	if(current_glide_host)
+		UnregisterSignal(current_glide_host, list(COMSIG_QDELETING, COMSIG_MOVABLE_UPDATE_GLIDE_SIZE))
+		current_glide_host = null
+	if(!QDELETED(new_glide_host) && new_glide_host != host)
+		RegisterSignal(new_glide_host, COMSIG_QDELETING, PROC_REF(reset_glide_host))
+		RegisterSignal(new_glide_host, COMSIG_MOVABLE_UPDATE_GLIDE_SIZE, PROC_REF(update_glide_size))
+		current_glide_host = new_glide_host
+		my_movable.set_glide_size(new_glide_host.glide_size)
+	else
+		my_movable.set_glide_size(initial(my_movable.glide_size))
+
+/datum/proximity_monitor/advanced/bubble/proc/reset_glide_host(datum/source)
+	SIGNAL_HANDLER
+	if(isnull(source) || source == current_glide_host)
+		set_glide_host(get_atom_on_turf(host))
```

Comment:
qdeleting seems like the wrong thing to trigger this, it seems like you effectively want `/datum/component/connect_containers` with glide_size here